src/windows/common/WSLAContainerLauncher.cpp

```diff
@@ -0,0 +1,74 @@
+/*++
+
+Copyright (c) Microsoft. All rights reserved.
+
+Module Name:
+
+    WSLAContainerLauncher.cpp
+
+Abstract:
+
+    This file contains the implementation for WSLAContainerLauncher.
+
+--*/
+#include "WSLAContainerLauncher.h"
+
+using wsl::windows::common::ClientRunningWSLAProcess;
+using wsl::windows::common::RunningWSLAContainer;
+using wsl::windows::common::WSLAContainerLauncher;
+
+RunningWSLAContainer::RunningWSLAContainer(wil::com_ptr<IWSLAContainer>&& Container, std::vector<WSLA_PROCESS_FD>&& fds) :
+    m_container(std::move(Container)), m_fds(std::move(fds))
+{
+}
+
+IWSLAContainer& RunningWSLAContainer::Get()
+{
+    return *m_container;
+}
+
+WSLA_CONTAINER_STATE RunningWSLAContainer::State()
+{
+    WSLA_CONTAINER_STATE state{};
+    THROW_IF_FAILED(m_container->GetState(&state));
+    return state;
+}
+
+ClientRunningWSLAProcess RunningWSLAContainer::GetInitProcess()
+{
+    wil::com_ptr<IWSLAProcess> process;
+    THROW_IF_FAILED(m_container->GetInitProcess(&process));
+
+    return ClientRunningWSLAProcess{std::move(process), std::move(m_fds)};
+}
+
+WSLAContainerLauncher::WSLAContainerLauncher(
+    const std::string& Image,
```

Comment:
Seems like we're doing a lot of string construction by passing these as std::string instead of PCSTR